src/config.py

```diff
@@ -40,6 +40,12 @@ def get_logs_dir(self):
     def get_repos_dir(self):
         return self.config["STORAGE"]["REPOS_DIR"]
     
+    def get_openai_base_url(self):
+        return self.config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
+    
+    def get_search_engine(self):
+        return self.config["USER_PREFS"]["SEARCH_ENGINE"]
```

Comment:
**Optimization Issue**: Direct access to configuration details in methods can lead to tight coupling and difficulties in managing configuration changes. <br> **Fix**: Encapsulate configuration access within a dedicated method or class that provides configuration data, improving maintainability and scalability. <br> **Code Suggestion**: 
 ```
 -        return self.config["USER_PREFS"]["SEARCH_ENGINE"]
 +        return self.get_user_preference("SEARCH_ENGINE")

 +    def get_user_preference(self, key):
 +        return self.config["USER_PREFS"].get(key, "default_value")
 ```